File: apollo-runtime/src/main/java/com/apollographql/apollo/internal/field/MapFieldValueResolver.java
package com.apollographql.apollo.internal.field;

import com.apollographql.apollo.api.ResponseField;

import java.util.Map;

public final class MapFieldValueResolver implements FieldValueResolver<Map<String, Object>> {

  @SuppressWarnings("unchecked") @Override public <T> T valueFor(Map<String, Object> map, ResponseField field) {
    return (T) map.get(field.responseName());
  }
}

File: apollo-runtime/src/main/java/com/apollographql/apollo/internal/field/MapFieldValueResolver.kt
package com.apollographql.apollo.internal.field

import com.apollographql.apollo.api.ResponseField
import com.apollographql.apollo.api.internal.FieldValueResolver

class MapFieldValueResolver : FieldValueResolver<Map<String, Any>> {
  override fun <T> valueFor(map: Map<String, Any>, field: ResponseField): T? {
    return map[field.responseName] as T?
  }
}